src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp

```diff
@@ -41,4 +66,305 @@ namespace TerminalAppLocalTests
         VERIFY_ARE_NOT_EQUAL(oldFontSize, newFontSize);
     }
 
+    void SettingsTests::ValidateProfilesExist()
+    {
+        const std::string settingsWithProfiles{ R"(
+        {
+            "profiles": [
+                {
+                    "name" : "profile0"
+                }
+            ]
+        })" };
+
+        const std::string settingsWithoutProfiles{ R"(
+        {
+            "defaultProfile": "{6239a42c-1de4-49a3-80bd-e8fdd045185c}"
+        })" };
+
+        const std::string settingsWithEmptyProfiles{ R"(
+        {
+            "profiles": []
+        })" };
+
+        {
+            // Case 1: Good settings
+            const auto settingsObject = VerifyParseSucceeded(settingsWithProfiles);
+            auto settings = CascadiaSettings::FromJson(settingsObject);
+            settings->_ValidateProfilesExist();
+        }
+        {
+            // Case 2: Bad settings
+            const auto settingsObject = VerifyParseSucceeded(settingsWithoutProfiles);
+            auto settings = CascadiaSettings::FromJson(settingsObject);
+            try
+            {
+                settings->_ValidateProfilesExist();
+            }
+            catch (const ::TerminalApp::SettingsException ex)
+            {
+                VERIFY_IS_TRUE(ex.Error() == ::TerminalApp::SettingsLoadErrors::NoProfiles);
+            }
+            catch (...)
+            {
+                VERIFY_FAIL(L"Expected a SettingsLoadErrors::NoProfiles, but caught another exception type");
+            }
+        }
+        {
+            // Case 3: Bad settings
+            const auto settingsObject = VerifyParseSucceeded(settingsWithEmptyProfiles);
+            auto settings = CascadiaSettings::FromJson(settingsObject);
+            try
+            {
+                settings->_ValidateProfilesExist();
+            }
+            catch (const ::TerminalApp::SettingsException ex)
+            {
+                VERIFY_IS_TRUE(ex.Error() == ::TerminalApp::SettingsLoadErrors::NoProfiles);
+            }
+            catch (...)

```

Comment:
Unless we know that there are exceptions which are not subclasses of std::exception my recommendation would be to catch std::exception instead (which allows printing what())

Reply:
Also doesn't catch(...) interfere with the test framework? Don't most test frameworks use exception to signal assertion failures?

Reply:
Shockingly, catch(...) doesn't seem to mess with the testing at all. So that's good!

I might actually just pull that handler out and let TAEF do the work for us there